src/main/java/com/ecommerce/ECommerceApp/exception/InvalidEmailException.java

```diff
@@ -0,0 +1,10 @@
+package com.ecommerce.ECommerceApp.exception;
+
+/**
+ * Throws an exception error when invalid email address is provided.
+ */
+public class InvalidEmailException extends RuntimeException {
+    public InvalidEmailException(String message) {
+        super(message);
+    }
+}
```

Comment:
I introduced a custom exception class to allow for more clear and targeted error handling inside the program. This will help make things easier to debug and provide a more clear understanding of what happened. 

Source: https://www.baeldung.com/java-clean-code